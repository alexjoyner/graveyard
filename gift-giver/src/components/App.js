import React, {Component} from 'react';

export default class App extends Component {
    render() {
        return (
            <div>
                <h1>Gift Giver</h1>
            </div>
        )
    }
}