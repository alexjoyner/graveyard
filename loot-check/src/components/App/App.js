import React, {Component} from 'react';
import Wallet from '../Wallet/Wallet';
import Loot from '../Loot/Loot';

export default class App extends Component {
    render() {
        return (
            <div>
                <h2>Loot Check</h2>
                <hr/>
                <Wallet/>
                <Loot/>
            </div>
        )
    }
}