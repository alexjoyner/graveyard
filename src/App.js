import React, { Component } from 'react';
import {connect} from 'react-redux';
import {
  Header, colors, SideBarPage, ListItem, 
  ListHeader, CenteredContent, Notification, Button} from 'ro-component-library';
import {GaugeBlockArray} from './organisms/GaugeBlockArray';
import {AuthModal} from './organisms/AuthModal';
import { HistoricalGraphModal } from './organisms/HistoricalGraphModal';

const SideBarContent = ({ roomCreds }) => (
  <div>
    <ListHeader>Rooms</ListHeader>
    <CenteredContent>
      <ListItem>Demo Room</ListItem>
      <ListItem color={colors.primaryLight}>+ Add Room</ListItem>
    </CenteredContent>
  </div>
)
class NotificationTest extends Component {
  constructor(props) {
    super(props);
    this.state = {
      notifShown: true,
    };
  }
  toggleModal() {
    this.setState({
      notifShown: !this.state.notifShown,
    });
  }
  render() {
    return (
      <div>
        <Notification shown={this.state.notifShown} onClose={() => this.toggleModal()}>
          <h3>Hello World</h3>
        </Notification>
        <Button primary onClick={() => this.toggleModal()} > Modal {this.state.notifShown ? 'shown' : 'hidden'} </Button>
      </div>
    );
  }
}
class App extends Component {
  render() {
    return this.props.loggedIn? (
      <div style={{marginTop: '80px'}}>
        <NotificationTest />
        <Header 
          color={colors.dark} 
          height={'67px'} 
          fontSize={'25px'} 
          logoText={'Dashboard Demo'} 
          sticky />
        <SideBarPage sideBarContents={<SideBarContent {...this.props} />} >
          <GaugeBlockArray></GaugeBlockArray>
        </SideBarPage>
        {(this.props.modalShown)?(
          <HistoricalGraphModal modalData={this.props.modalData}/>
        ):null}
    </div>) : (
      <AuthModal></AuthModal>
    );
  }
}
const mapStateToProps = (state) => {
  return {
    ...state.HistoricalGraphModalReducer,
    ...state.AuthModalReducer
  }
}
export default connect(mapStateToProps)(App);