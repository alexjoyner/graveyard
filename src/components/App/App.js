import React from 'react';
import {Knight} from "../Knight/Knight";

const App = () => {
    return (
        <div>
            <h1>Hello World</h1>
            <Knight/>
        </div>
    )
};

export default App;