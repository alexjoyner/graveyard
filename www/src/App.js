import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getGroups } from './organisms/SideBarContent/utils/getGroups';
import { setGroups } from './organisms/SideBarContent/actions';
import { DashBoard } from './pages/DashBoard';

export class App extends Component {
  async componentDidMount() {
    const groups = await getGroups();
    this.props.setGroups(groups);
  }

  render() {
    return (
      <DashBoard />
    );
  }
}
export default connect(null, {
  setGroups,
})(App);
