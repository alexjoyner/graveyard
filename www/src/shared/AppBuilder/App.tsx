import { combineReducers } from './combineReducers';
import { StoreContext } from './storeContext';
import React, { ReactNode, useReducer } from 'react';
import { ReducersObject, GenericFeatures } from './types';
import { Reducer, ReducersMapObject, AnyAction } from 'redux';
import { FeaturesContext } from './featuresContext';

export interface App<F = GenericFeatures> {
	getFeatures(): F;
	getReducers(): ReducersMapObject<any, AnyAction>;
	addFeatures<App>(Featurea: { new(...args: any[]): App }[]): App;
	Run({
		reducers,
		features,
	}: {
		reducers?: ReducersMapObject<any, AnyAction>,
		features?: F
	}): ReactNode;
}


//  Base DashBoard
export abstract class BasicApp<F = GenericFeatures> implements App<F> {
	private features: F;
	private reducers: ReducersMapObject<any, AnyAction>;
	private rootContent: ReactNode;
	private defaultState: Object;
	constructor(
		features: F,
		reducers: ReducersMapObject<any, AnyAction>,
		rootContent: ReactNode,
		defaultState?: Object
	) {
		this.features = features;
		this.reducers = reducers;
		this.rootContent = rootContent;
		this.defaultState = defaultState || {};
	}
	getFeatures(): F {
		return this.features;
	}
	getReducers(): ReducersMapObject<any, AnyAction> {
		return this.reducers;
	}
	addFeatures<App>(Features: { new(...args: any[]): App }[]): App {
		let newApp = new Features[0](this);
		Features.map((Feature, indx) => {
			if (indx > 0) {
				newApp = new Feature(newApp);
			}
		})
		return newApp;
	}
	private Build({
		reducers,
		features,
		rootContent,
		defaultState,
	}: {
		reducers: ReducersMapObject<any, AnyAction>,
		features: F,
		rootContent: ReactNode,
		defaultState: Object
	}) {
		const rootReducer = combineReducers(reducers);
		const initialState = rootReducer({}, { type: '__INIT__' });
		const [state, dispatch] = useReducer(rootReducer, initialState);
		const finalState = { ...defaultState, ...state };
		return (
			<StoreContext.Provider value={[finalState, dispatch]} >
				<FeaturesContext.Provider value={features}>
					{rootContent}
				</FeaturesContext.Provider>
			</StoreContext.Provider>
		);
	}
	Run({
		reducers,
		features,
	}: {
		reducers?: ReducersMapObject<any, AnyAction>,
		features?: F
	}) {
		return <this.Build
			reducers={reducers || this.getReducers()}
			features={features || this.getFeatures()}
			rootContent={this.rootContent}
			defaultState={this.defaultState} />
	}
}