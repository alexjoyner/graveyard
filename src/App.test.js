import React from 'react';
import { shallow } from 'enzyme';
import { App } from './App';

const mockFetch = data => jest.fn().mockImplementation(() => Promise.resolve({
  status: 200,
  json: () => data,
}));

describe('App component', () => {
  let component;
  beforeEach(() => {
    window.fetch = mockFetch('test');
    const props = {
      setGroups: jest.fn(),
    };
    component = shallow(<App {...props} />);
  });
  it('Should render without exploding', () => {
    expect(component).toBeDefined();
  });
  it('Should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
