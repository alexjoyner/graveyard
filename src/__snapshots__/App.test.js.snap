// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown />,
        <Header
          color="dark"
          fontSize="25px"
          height="67px"
          logoText="Dashboard Demo"
          sticky={true}
>
          <FeedbackButton />
</Header>,
        <SideBarPage
          sideBarContents={<Connect(BaseSideBarContent) />}
>
          <Connect(BaseGaugeBlockArray) />
</SideBarPage>,
        <Connect(BaseHistoricalGraphModal)
          modalData={undefined}
/>,
      ],
      "style": Object {
        "marginTop": "80px",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <FeedbackButton />,
          "color": "dark",
          "fontSize": "25px",
          "height": "67px",
          "logoText": "Dashboard Demo",
          "sticky": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(BaseGaugeBlockArray) />,
          "sideBarContents": <Connect(BaseSideBarContent) />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "modalData": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown />,
          <Header
            color="dark"
            fontSize="25px"
            height="67px"
            logoText="Dashboard Demo"
            sticky={true}
>
            <FeedbackButton />
</Header>,
          <SideBarPage
            sideBarContents={<Connect(BaseSideBarContent) />}
>
            <Connect(BaseGaugeBlockArray) />
</SideBarPage>,
          <Connect(BaseHistoricalGraphModal)
            modalData={undefined}
/>,
        ],
        "style": Object {
          "marginTop": "80px",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <FeedbackButton />,
            "color": "dark",
            "fontSize": "25px",
            "height": "67px",
            "logoText": "Dashboard Demo",
            "sticky": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(BaseGaugeBlockArray) />,
            "sideBarContents": <Connect(BaseSideBarContent) />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "modalData": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
